packages/compass-connections/src/stores/connections-store.tsx

```diff
@@ -211,13 +236,53 @@ export function useConnections({
   );
   const { activeConnectionId } = state;
 
+  const getActiveAndInactiveCount = useCallback(() => {
+    const total = favoriteConnections.length + recentConnections.length;
+    const active =
+      connectionsManager.getConnectionIdsByStatus(ConnectionStatus.Connected)
+        .length || 0;
+    return { active, inactive: total - active };
+  }, [favoriteConnections, recentConnections, connectionsManager]);
+
+  const findConnectionInfo = useCallback(
+    (connectionId: ConnectionInfo['id']) => {
+      return [...favoriteConnections, ...recentConnections].find(
+        ({ id }) => id === connectionId
+      );
+    },
+    [favoriteConnections, recentConnections]
+  );
+
   const saveConnectionInfo = useCallback(
-    async (
-      connectionInfo: RecursivePartial<ConnectionInfo> &
-        Pick<ConnectionInfo, 'id'>
-    ) => {
+    async ({
+      fullConnectionInfo,
+      partialConnectionInfo,
+    }:
+      | {
+          fullConnectionInfo?: undefined;
+          partialConnectionInfo: RecursivePartial<ConnectionInfo> &
+            Pick<ConnectionInfo, 'id'>;
+        }
+      | {
+          fullConnectionInfo: ConnectionInfo;
+          partialConnectionInfo?: undefined;
+        }) => {
+      const connectionInfo = fullConnectionInfo || partialConnectionInfo;
       try {
-        await saveConnection(connectionInfo);
+        if (fullConnectionInfo) {
+          const isNewConnection = !findConnectionInfo(fullConnectionInfo.id);
```

Comment:
can't you rely on the fact that `id` is set to know if is new, but also, isn't the callsite aware of this?

Reply:
I don't follow the question - did you mean that the id is not set for a new connection? if so, this is actually incorrect, the id is created when the form for a new connection is open.

Reply:
Ah ok, i thought is something that is done on save